### PR TITLE
[tomlplusplus] update to 2.4.0

### PR DESCRIPTION
**Pull request by marzer**
**Describe the pull request**

- #### What does your PR fix?  
Updates the toml++ port to the v2.4.0 release.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
No changes here.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
Yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
Yes.

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/tomlplusplus/CONTROL [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Source: tomlplusplus
-Version: 2.3.0
+Version: 2.4.0
 Port-Version: 1
```
Comment (JonLiu1993) [inline ports/tomlplusplus/CONTROL:3]: @marzer ,Please delete "Port-Version", thanks

Comment (marzer) [reply ports/tomlplusplus/CONTROL:3]: Um, sure? That was put there by `./vcpkg x-add-version --all`, which the PR template asked me to run...

**Comment by JonLiu1993** [comment]
@marzer, please run the command "./vcpkg x-add-version tomlplusplus --overwrite-version" and submit again, thanks

File: versions/t-/tomlplusplus.json [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "95397531895566a6af15644be1ea9e228795c02d",
+      "version-string": "2.4.0",
+      "port-version": 0
+    },
+    {
+      "git-tree": "81478bfd62b4e2f180eadf83f1848d6e8db97cef",
+      "version-string": "2.4.0",
+      "port-version": 1
+    },
```
Comment (JonLiu1993) [inline versions/t-/tomlplusplus.json:12]: When upgrading the port, we only need to add "version" once, please delete this

Comment (marzer) [reply versions/t-/tomlplusplus.json:12]: Ok, sure, but it really seems like you need to improve your PR instructions for port upgrades, or improve the tooling to not require all of this JSON-related intervention. I simply followed the instructions...

Comment (JonLiu1993) [reply versions/t-/tomlplusplus.json]: > Ok, sure, but it really seems like you need to improve your PR instructions for port upgrades, or improve the tooling to not require all of this JSON-related intervention. I simply followed the instructions...

Thank you, we will consider your suggestion

**Review by JonLiu1993 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks for the PR and another release of your awesome library :)

**Pull request merged**